src/treehash.jl

```diff
@@ -0,0 +1,174 @@
+# The code in this file is taken from Pkg.GitTools
```

Comment:
Could you add a github permalink here that refers to the concrete repository, file, and hash this is taken from? My main point for this is to make it easier in the future to check for possible bugfixes to the upstream code and allowing it to copy an updated version of the code over to this file